src/ocean/task/TaskPool.d

```diff
@@ -274,3 +315,47 @@ unittest
     test!("==")(RecycleTask.recycle_count, 2,
         "RecycleTask.recycle was not called the correct number of times");
 }
+
+unittest
+{
+    // Test for waiting until all running tasks of a task pool finish executing.
+
+    static class AwaitTask : Task
+    {
+        static int value;
+
+        public void copyArguments ( )
+        {
+        }
+
+        override public void run ( )
+        {
+            value = 1;
+            .wait(1);
+            value = 2;
+        }
+    }
+
+    class MainTask : Task
+    {
+        TaskPool!(AwaitTask) my_task_pool;
+
+        public this ( )
+        {
+            this.my_task_pool = new TaskPool!(AwaitTask);
+        }
+
+        override protected void run ( )
+        {
+            this.my_task_pool.start();
```

Comment:
Running `start` 2-3 times here while keeping rest of test code the same will be a better check in my opinion (the one that ensures that _all_ tasks are waited for)